lib/robots/MiioValetudoRobot.js

```diff
@@ -182,31 +182,29 @@ class MiioValetudoRobot extends ValetudoRobot {
         let localSecret = this.implConfig.localSecret;
 
         if (!localSecret) {
-            let line;
-
             Logger.trace("Trying to read token file at " + this.tokenFilePath);
             try {
-                line = fs.readFileSync(this.tokenFilePath);
+                localSecret = fs.readFileSync(this.tokenFilePath);
             } catch (e) {
-                Logger.debug("cannot read", this.tokenFilePath, e);
-            }
-            if (line && line.length >= 32) {
-                // For local development, people might put in the hex representation of the token.
-                // Make this work too.
-                return Buffer.from(line.toString().slice(0, 32), "hex");
+                Logger.debug("Cannot read token file at", this.tokenFilePath, "Local communication will fail", e);
+                return Buffer.alloc(16);
             }
+        } else {
+            Logger.trace("Parsing localSecret from config");
+        }
 
-            if (line && line.length >= 16) {
-                return line.slice(0, 16);
-            }
-
-            Logger.error("Unable to read token file. Local communication will fail");
-
-            return Buffer.alloc(16);
+        if (localSecret && localSecret.length >= 32) {
+            // For local development, people might put in the hex representation of the token.
+            // Make this work too.
+            return Buffer.from(localSecret.toString().slice(0, 32), "hex");
+        }
 
-        } else {
-            return Buffer.from(localSecret, "hex");
+        if (localSecret && localSecret.length >= 16) {
+            return localSecret.slice(0, 16);
         }
+
+        Logger.debug("Invalid localSecret with length " + localSecret.length);
+        return Buffer.alloc(16);
     }
 
```

Comment:
I refactored this code because the Viomi provides the local secret in `/etc/miio/device.token` and it is not hex-encoded.
This recycles the same code used when run on the vacuum for parsing the value provided in the configuration.